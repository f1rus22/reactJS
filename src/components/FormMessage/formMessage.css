.formMessage {
    position: fixed;
    bottom: 30px; 
    left: 0;
    right: 0;
    margin: 0 auto;
    width: 600px;
    background-color:rgb(248, 248, 248);
    box-shadow: 5px 10px 50px rgb(16 112 177 / 20%);
    display:grid;
    grid-template-columns: 100px 1fr;
    grid-row-gap: 10px;

    padding: 10px;
    box-sizing: border-box;
    border-radius: 20px;
    
}

button{
    width: 200px;
    padding: 10px;
    background-color: rgb(107, 181, 241);
    color: white;
    border: none;
    border-radius: 10px;

}

label{
    text-align: left;
}