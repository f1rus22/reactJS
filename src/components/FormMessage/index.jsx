import { Button, Input } from "@mui/material";
import SendIcon from "@mui/icons-material/Send";
import React from "react";
import { useState, useCallback } from "react";
import faker from "faker";
import { Box } from "@mui/system";
import { useSelector } from "react-redux";
import { selectName } from "../../store/Profile/selectors";

export default function FormMessage({ loadMessage }) {
  const [value, setValue] = useState("");

  const profileName = useSelector(selectName);

  const changeFormValue = useCallback((e) => {
    setValue(e.target.value);
  }, []);

  const handleSubmit = (e) => {
    e.preventDefault();
    loadMessage({
      author: profileName,
      text: value,
      id: faker.datatype.number(),
    });
    setValue("");
  };

  return (
    <form onSubmit={handleSubmit}>
      <Box sx={{ display: "flex", flexDirection: "row" }}>
        <Input
          fullWidth="true"
          autoFocus
          value={value}
          onChange={changeFormValue}
          type="text"
        />
        <Button
          variant="contained"
          style={{ height: "37px" }}
          onClick={handleSubmit}
        >
          <SendIcon />
        </Button>
      </Box>
    </form>
  );
}
