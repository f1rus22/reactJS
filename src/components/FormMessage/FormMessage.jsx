import React, { useEffect, useRef } from "react";
import "./formMessage.css";
import { Button, TextField, Box } from "@material-ui/core";

export default function FormMessage({
  handleMessFormSubmit,
  setIsTyping,
  message,
  setMessage,
}) {
  const ref = useRef(null);
  useEffect(() => {
    ref.current?.focus();
  });

  return (
    <Box
      className="formMessage"
      sx={{ dispplay: "flex", height: "80px" }}
      onSubmit={handleMessFormSubmit}
    >
      <TextField
        sx={{ width: "auto" }}
        label="Message"
        variant="outlined"
        type="text"
        name="text"
        id="text"
        inputRef={ref}
        value={message}
        onChange={(event) => {
          setIsTyping(true);
          setMessage(event.target.value);
        }}
      />
      <Button
        onClick={handleMessFormSubmit}
        variant="contained"
        color="primary"
      >
        Send
      </Button>
    </Box>
  );
}
