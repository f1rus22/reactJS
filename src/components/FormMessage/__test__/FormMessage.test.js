import { render } from '@testing-library/react';
import FormMessage from "..";

it('snapshot component Form', () => {
    const component = render(<FormMessage />)
    expect(component).toMatchSnapshot();
})