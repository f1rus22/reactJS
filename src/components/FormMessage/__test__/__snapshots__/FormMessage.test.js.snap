// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`snapshot component Form 1`] = `
Object {
  "asFragment": [Function],
  "baseElement": <body>
    <div>
      <form>
        <div
          class="MuiBox-root css-1xhj18k"
        >
          <div
            class="MuiInput-root MuiInput-underline MuiInputBase-root MuiInputBase-colorPrimary Mui-focused css-a88p61-MuiInputBase-root-MuiInput-root"
          >
            <input
              class="MuiInput-input MuiInputBase-input css-1x51dt5-MuiInputBase-input-MuiInput-input"
              type="text"
              value=""
            />
          </div>
          <button
            class="MuiButton-root MuiButton-contained MuiButton-containedPrimary MuiButton-sizeMedium MuiButton-containedSizeMedium MuiButtonBase-root  css-sghohy-MuiButtonBase-root-MuiButton-root"
            style="height: 37px;"
            tabindex="0"
            type="button"
          >
            <svg
              aria-hidden="true"
              class="MuiSvgIcon-root MuiSvgIcon-fontSizeMedium css-i4bv87-MuiSvgIcon-root"
              data-testid="SendIcon"
              focusable="false"
              viewBox="0 0 24 24"
            >
              <path
                d="M2.01 21 23 12 2.01 3 2 10l15 2-15 2z"
              />
            </svg>
            <span
              class="MuiTouchRipple-root css-8je8zh-MuiTouchRipple-root"
            />
          </button>
        </div>
      </form>
    </div>
  </body>,
  "container": <div>
    <form>
      <div
        class="MuiBox-root css-1xhj18k"
      >
        <div
          class="MuiInput-root MuiInput-underline MuiInputBase-root MuiInputBase-colorPrimary Mui-focused css-a88p61-MuiInputBase-root-MuiInput-root"
        >
          <input
            class="MuiInput-input MuiInputBase-input css-1x51dt5-MuiInputBase-input-MuiInput-input"
            type="text"
            value=""
          />
        </div>
        <button
          class="MuiButton-root MuiButton-contained MuiButton-containedPrimary MuiButton-sizeMedium MuiButton-containedSizeMedium MuiButtonBase-root  css-sghohy-MuiButtonBase-root-MuiButton-root"
          style="height: 37px;"
          tabindex="0"
          type="button"
        >
          <svg
            aria-hidden="true"
            class="MuiSvgIcon-root MuiSvgIcon-fontSizeMedium css-i4bv87-MuiSvgIcon-root"
            data-testid="SendIcon"
            focusable="false"
            viewBox="0 0 24 24"
          >
            <path
              d="M2.01 21 23 12 2.01 3 2 10l15 2-15 2z"
            />
          </svg>
          <span
            class="MuiTouchRipple-root css-8je8zh-MuiTouchRipple-root"
          />
        </button>
      </div>
    </form>
  </div>,
  "debug": [Function],
  "findAllByAltText": [Function],
  "findAllByDisplayValue": [Function],
  "findAllByLabelText": [Function],
  "findAllByPlaceholderText": [Function],
  "findAllByRole": [Function],
  "findAllByTestId": [Function],
  "findAllByText": [Function],
  "findAllByTitle": [Function],
  "findByAltText": [Function],
  "findByDisplayValue": [Function],
  "findByLabelText": [Function],
  "findByPlaceholderText": [Function],
  "findByRole": [Function],
  "findByTestId": [Function],
  "findByText": [Function],
  "findByTitle": [Function],
  "getAllByAltText": [Function],
  "getAllByDisplayValue": [Function],
  "getAllByLabelText": [Function],
  "getAllByPlaceholderText": [Function],
  "getAllByRole": [Function],
  "getAllByTestId": [Function],
  "getAllByText": [Function],
  "getAllByTitle": [Function],
  "getByAltText": [Function],
  "getByDisplayValue": [Function],
  "getByLabelText": [Function],
  "getByPlaceholderText": [Function],
  "getByRole": [Function],
  "getByTestId": [Function],
  "getByText": [Function],
  "getByTitle": [Function],
  "queryAllByAltText": [Function],
  "queryAllByDisplayValue": [Function],
  "queryAllByLabelText": [Function],
  "queryAllByPlaceholderText": [Function],
  "queryAllByRole": [Function],
  "queryAllByTestId": [Function],
  "queryAllByText": [Function],
  "queryAllByTitle": [Function],
  "queryByAltText": [Function],
  "queryByDisplayValue": [Function],
  "queryByLabelText": [Function],
  "queryByPlaceholderText": [Function],
  "queryByRole": [Function],
  "queryByTestId": [Function],
  "queryByText": [Function],
  "queryByTitle": [Function],
  "rerender": [Function],
  "unmount": [Function],
}
`;
